python/mxnet/gluon/estimator/estimator.py

```diff
@@ -241,16 +240,35 @@ def fit(self, train_data,
             from a data batch and load into contexts(devices)
         """
 
-
         self.epochs = epochs
-        if not batch_size:
-            batch_size = 32 * len(self.context)
+        if isinstance(train_data, gluon.data.DataLoader):
+            num_batches = len(train_data)
+            total_samples = len(train_data._dataset)
+            if total_samples == 0:
+                raise ValueError("DataLoader is Empty. Please refer to gluon.data.DataLoader "
+                                 "for more detail")
+            for dt, _ in train_data:
```

Comment:
Can this be condensed to 1 line? How about `np.ceil(len(train_data._dataset)/len(train_data))`

Reply:
That won't work. Here is an example:

Total number of samples: 270 (`len(train_data._dataset)`)
Batch size(In dataLoader): 32
Number of batches from dataLoader: 9 (`len(train_data)`) [32 * 8 + 14]

`np.ceil(len(train_data._dataset)/len(train_data))`: 30.0 (Correct batch size is 32)

Reply:
yes that won't work, thanks for clarifying.